pkg/controller/tenant/tenant_rbac_test.go

```diff
@@ -0,0 +1,102 @@
+package tenant
+
+import (
+	"testing"
+
+	synv1alpha1 "github.com/projectsyn/lieutenant-operator/pkg/apis/syn/v1alpha1"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+func assertManagedByLabel(t *testing.T, obj metav1.ObjectMeta, value string) {
+	v, ok := obj.Labels["app.kubernetes.io/managed-by"]
+	assert.True(t, ok, "the label `app.kubernetes.io/managed-by` is not set")
+	assert.Equal(t, value, v)
+}
+
+func assertOwnerReference(t *testing.T, obj metav1.Object, owner *synv1alpha1.Tenant) {
+	refs := obj.GetOwnerReferences()
+	require.Len(t, refs, 1)
+	assert.Equal(t, owner.GetName(), refs[0].Name)
+}
+
+func newReconcileTenant() *ReconcileTenant {
+	s := scheme.Scheme
+	objs := []runtime.Object{
+		&synv1alpha1.Tenant{},
+	}
+	s.AddKnownTypes(synv1alpha1.SchemeGroupVersion, objs...)
+	c := fake.NewFakeClientWithScheme(s, objs...)
+	return &ReconcileTenant{client: c, scheme: s}
+}
+
+func TestReconcileTenant_NewServiceAccount(t *testing.T) {
+	tenant := &synv1alpha1.Tenant{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "my-tenant",
+			Namespace: "my-namespace",
+		},
+	}
+
+	r := newReconcileTenant()
+	sa, err := r.NewServiceAccount(tenant)
+	require.NoError(t, err)
+
+	assert.Equal(t, tenant.Name, sa.Name)
+	assert.Equal(t, tenant.Namespace, sa.Namespace)
+	assertManagedByLabel(t, sa.ObjectMeta, "lieutenant")
+	assertOwnerReference(t, sa, tenant)
+}
+
+func TestReconcileTenant_NewRole(t *testing.T) {
+	tenant := &synv1alpha1.Tenant{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "my-tenant",
+			Namespace: "my-namespace",
+		},
+	}
```

Comment:
This piece seems to be used more often. Maybe add a helper `newTenant()` in the test suite ?

Reply:
I have had that tought too. But then, sometimes more than just the object meta is needed. To me it felt like over apstracting, so I left it as is.